Detectors/TPC/monitor/macro/RunFindAdcError.C

```diff
@@ -0,0 +1,171 @@
+            std::get<1>(err),
+            std::get<2>(err));
+      }
+    }
+  }
+//  LOG(INFO) << reader->getEventInfo(0)->at(0).path << " done" << FairLogger::endl;
+}
+
+//__________________________________________________________________________
+void RunFindAdcError(int run_min, int run_max)
+{
+  std::string DATADIR = "/local/data/tpc-beam-test-2017";
+  FairLogger *logger = FairLogger::GetLogger();
+  logger->SetLogVerbosityLevel("LOW");
+  logger->SetLogScreenLevel("INFO");
+
+
+  // ===========================================================================
+  // Preparing File Infos 
+  // ===========================================================================
+  std::vector<std::string> fileInfos;
+  for (int r = run_min; r <= run_max; ++r) {
+    std::ostringstream ss;
+    ss << DATADIR << "/run" << std::setw(6) << std::setfill('0') << r << "/run" << std::setw(6) << std::setfill('0') << r;
+
+    fileInfos.emplace_back(ss.str()+"_trorc00_link00.bin:0:9");
+    fileInfos.emplace_back(ss.str()+"_trorc00_link01.bin:1:9");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link02.bin:0:10");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link03.bin:1:10");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link04.bin:0:11");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link05.bin:1:11");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link06.bin:2:11");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link07.bin:3:11");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link08.bin:2:12");
+//    fileInfos.emplace_back(ss.str()+"_trorc00_link09.bin:3:12");
+    fileInfos.emplace_back(ss.str()+"_trorc00_link10.bin:2:13");
+    fileInfos.emplace_back(ss.str()+"_trorc00_link11.bin:3:13");
+  }
+
+
+  // ===========================================================================
+  // Preparing the Readers 
+  // ===========================================================================
+  LOG(INFO) << "Create all Readers..." << FairLogger::endl;
+  std::vector<std::shared_ptr<RawReader>> RawReaders;
+  for (const auto &s : fileInfos) {
+    auto rawReader = new RawReader;
+    rawReader->addInputFile(s);
+    rawReader->setUseRawInMode3(true);
+    rawReader->setCheckAdcClock(true);
+
+    // keep only raw data
+    if (rawReader->getEventInfo(0)->at(0).header.dataType != 2)
+      RawReaders.push_back(std::shared_ptr<RawReader>(rawReader));
+    else
+      LOG(INFO) << "\tDrop Reader of file " << rawReader->getEventInfo(0)->at(0).path << " because of readout mode 2." << FairLogger::endl;
+  }
+  LOG(INFO) << "... done" << FairLogger::endl;
+
+
+
+  // ===========================================================================
+  // Loop through the Readers to find ADC errors
+  // ===========================================================================
+  LOG(INFO) << "Loop through Readers..." << FairLogger::endl;
+  std::vector<std::thread> threads;
+  std::vector<std::vector<Result>> results(RawReaders.size());
+  int i = 0;
```

Comment:
Since I am seeing threads here, I am wondering who is using this macro? Is it a utility that is used by developers or will it run as part of the final data processing?